lib/Target/DebugInfo/EmitHGLDD.cpp

```diff
@@ -531,23 +538,46 @@ void FileEmitter::emitModule(JOStream &json, DIModule *module) {
     for (auto *instance : module->instances)
       emitInstance(json, instance);
   });
-  json.objectEnd();
 }
 
 /// Emit the debug info for a `DIInstance`.
 void FileEmitter::emitInstance(JOStream &json, DIInstance *instance) {
   json.objectBegin();
+
+  // Emit the instance name (HGL and optionally HDL if different).
   json.attribute("name", instance->name.getValue());
-  auto verilogName = getVerilogInstanceName(*instance);
-  if (verilogName != instance->name)
-    json.attribute("hdl_obj_name", verilogName.getValue());
-  json.attribute("obj_name", instance->module->name.getValue()); // HGL
-  json.attribute("module_name",
-                 getVerilogModuleName(*instance->module).getValue()); // HDL
+  if (!instance->module->isInline) {
+    auto verilogName = getVerilogInstanceName(*instance);
+    if (verilogName != instance->name)
+      json.attribute("hdl_obj_name", verilogName.getValue());
+  }
+
+  // Emit the module name (HGL and HDL), unless this is an inline scope.
+  if (!instance->module->isInline) {
```

Comment:
Any reason this if block cannot be pulled into the last one ? Looks like same condition.

Reply:
Oh yeah thanks for catching this. :see_no_evil: